ledger/daml-on-sql/README.rst

```diff
@@ -0,0 +1,742 @@
+.. Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+DAML-on-SQL
+###########
+
+DAML-on-SQL is a PostgreSQL-based DAML ledger implementation.
+
+Setup PostgreSQL and run
+************************
+
+Before starting, you need to perform the following steps:
+
+- create an initially empty PostgresSQL database that DAML-on-SQL can access
+- have a database user for DAML-on-SQL that has authority to execute DDL operations
+
+This is because DAML-on-SQL manages its own database schema,
+applying migrations if necessary when upgrading versions.
+
+To specify the PostgreSQL instance you wish to connect, use the
+``--sql-backend-jdbcurl <value>`` command line option, where ``<value>``
+is a valid JDBC URL containing the username, password and database
+name to connect to.
+
+Here is an example for such a URL: ``jdbc:postgresql://localhost/test?user=fred&password=secret``
+
+Due to possible conflicts between the ``&`` character and various terminal
+shells, we recommend quoting the JDBC URL like so:
+
+.. code-block:: none
+
+  $ java -jar daml-on-sql-<version>.jar Main.dar --sql-backend-jdbcurl "jdbc:postgresql://localhost/test?user=fred&password=secret"
+
+If you're not familiar with JDBC URLs, see the PostgreSQL JDBC docs for more information: https://jdbc.postgresql.org/documentation/head/connect.html
+
+Architecture and availability
+*****************************
+
+Processes and components
+========================
+
+The core processes necessary to run a DAML-on-SQL deployment are:
+
+- the JVM-hosted standalone Ledger API server and
```

Comment:
I think we can probably refer to this as the "ledger server", as it's more than just the ledger API: it's the ledger itself, the ledger API, and the index.

Reply:
That is an implementation detail, I don't think we should commit to it in the docs.

Reply:
I think we should refer to the process as _DAML on SQL_ instead of _Ledger API Server_, _Participant Server_, or _Ledger Server_. All of the latter ones are implementation details.

Instead of `JVM-hosted` I think the docs could explicitly state which JVM version DAML on SQL is tested with and that this particular version is the minimum requirement and that other versions could work as well.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6850/commits/f57199d0811464e1f0c13d7b9bd3a98f5873cd7a

Reply:
Addressed by https://github.com/digital-asset/daml/pull/6850/commits/f57199d0811464e1f0c13d7b9bd3a98f5873cd7a